parser.y

```diff
@@ -3898,6 +3902,14 @@ Expression:
 			$$ = &ast.UnaryOperationExpr{Op: opcode.Not, V: $2}
 		}
 	}
+|	"MATCH" '(' ColumnNameList ')' "AGAINST" '(' StringLiteral FulltextSearchModifierOpt ')'
```

Comment:
Wait, the AGAINST part can be an arbitrary (string) expression, not just string literal.

```
mysql> select match(a) against (1234), match(a) against (now()), match(a) against (@@sql_mode in boolean mode) from x;
+-------------------------+--------------------------+-----------------------------------------------+
| match(a) against (1234) | match(a) against (now()) | match(a) against (@@sql_mode in boolean mode) |
+-------------------------+--------------------------+-----------------------------------------------+
|                       0 |                        0 |                                             0 |
|                       0 |                        0 |                                             0 |
+-------------------------+--------------------------+-----------------------------------------------+
2 rows in set (0.00 sec)
```

Reply:
Hm, it looks strange to query from variable though.

Reply:
I think it's better to support it for compatibility with MySQL. I guess const-evaluating the expression in TiDB isn't difficult.

Reply:
Agree, let me fix it.

Reply:
@kennytm